### ISSUE TITLE
SSDT Install is not reliable

### ISSUE THREAD

**Issue by efog** [Microsoft/vsts-image-generation#143]
We're running the VS2017 installation and can't get a reliable build since it fails during the SSDT install with this error:

> 2018-05-18T15:14:24.4276990Z     vhd: [15D8:15E0][2018-05-18T15:14:05]e000: Error 0x800707d3: Process returned error: 0x7d3
> 2018-05-18T15:14:24.4281468Z     vhd: [15D8:15E0][2018-05-18T15:14:05]e000: Error 0x800707d3: Failed to execute EXE package.
> 2018-05-18T15:14:24.4286353Z     vhd: [17C4:051C][2018-05-18T15:14:05]e000: Error 0x800707d3: Failed to configure per-machine EXE package.
> 2018-05-18T15:14:24.4292625Z     vhd: [17C4:051C][2018-05-18T15:14:05]i000: MainViewModel.OnPackageAction: Install Completed for package SQL Server Analysis Services (id: Microsoft.DataTools.AnalysisServices)
> 2018-05-18T15:14:24.4297271Z     vhd: [17C4:051C][2018-05-18T15:14:05]i319: Applied execute package: Microsoft.DataTools.AnalysisServices, result: 0x800707d3, restart: None
> 2018-05-18T15:14:24.4301953Z     vhd: [17C4:051C][2018-05-18T15:14:05]e000: Error 0x800707d3: Failed to execute EXE package.

This problem is also being reported by some others:

[https://stackoverflow.com/questions/46421619/ssdt-installation-issue-failed-to-execute-exe-package](https://stackoverflow.com/questions/46421619/ssdt-installation-issue-failed-to-execute-exe-package)

Using the same setup we've been able to generate an image so something has changed in the meantime or this component install could be at best described as un-reliable.

**Comment by bryanmacfarlane** [comment]
Also: https://github.com/Microsoft/vsts-image-generation/issues/132

**Comment by chrisrpatterson** [comment]
@efog you should also post something on developer community if there is a issue specifically with one component.  While we have posted these scripts and templates to help the community we do not own the install and support for every piece of software that might be installed.

**Comment by ignatz42** [comment]
@chrisrpatterson Many thanks for the work shared here. I am confused by your response. This repository is currently unbuildable and has been for at least the last month. The branches master(at ID 86684c70d7d460cd212a62911a2e465438f107e8), releases/1805.1, releases/1805, releases/1804 all fail to build. This problem seems to be happening because the vsts-image-generation repository always uses the latest version of Visual Studio as opposed to defining a specific version as-is the case with other tools like SSDT. The result of this policy is the current state of the repository. With this in mind, I hope you won't mind answering a few questions:

1. What source was used to produce the current Hosted 2017 Build Agents?  I have specific tests failing on my build agents based on releases/1804 created Apr 26, 2018 but working on the current Hosted 2017 agents. I'd very much like to be able to work out differences and fix my build agents but it seems all releases made in the last 7 months will fail this way.

2. Why isn't a specific version of Visual Studio specified in vsts-image-generation\images\win\scripts\Installers\Vs2017\Install-VS2017.ps1? Since this policy breaks once working source (ie release/1804) does it qualify as a bug?

**Comment by lokesh755** [comment]
@ignatz42 It's true that SSDT install is little unreliable at this point of time. But with the latest fix I committed it'll succeed most of the times. Also, I've created a [branch ](https://github.com/Microsoft/vsts-image-generation/tree/VS2017/15.6.7) which uses 15.6.7 version of VS2017 which is the same version installed on image generated on April 26, 2018.  The current problem SSDT install failures is introduced with VS 15.7.* . We were told that this will be fixed by next release of SSDT (15.7.0).
